sys/linux/init_images.go

```diff
@@ -64,9 +64,9 @@ func (zr *zeroReader) Read(p []byte) (n int, err error) {
 
 const imageMaxSize = 129 << 20
 
-func fixUpImageSegments(parsed *mountImageArgs) {
-	const maxImageSegments = 4096
-	parsed.filterSegments(func(i int, segment *mountImageSegment) bool {
+func fixUpImageSegments(parsed *mountImageArgs, fixStructure bool) error {
+	const maxImageSegments = 16784
```

Comment:
@hrutvik @melver FYI
As a side-effect of this change @nogikh noticed that some our syz_mount_image seeds were truncated b/c contained too many segments. Now they won't be truncated and any truncations in seeds will fail tests.

Reply:
Have to close this conversation, as otherwise it does not let me merge :(